inet/vibe/inet/url.d

```diff
@@ -418,6 +419,98 @@ struct URL {
 		return cast(NativePath)this.path;
 	}
 
+	/// Decode percent encoded triplets for unreserved or convert to uppercase
+	private string normalize_percent_encoding(scope const(char)[] input)
+	{
+		char[] normalized;
+
+		for (size_t i = 0; i < input.length; i++)
+		{
+			const char c = input[i];
+			if (c == '%')
+			{
+				if (input.length < i + 3)
+					assert(false, "Invalid percent encoding");
+				
+				char conv = cast(char) input[i + 1 .. i + 3].to!ubyte(16);
+				switch (conv)
+				{
+					case 'A': .. case 'Z':
+					case 'a': .. case 'z':
+					case '0': .. case '9':
+					case '-': case '.': case '_': case '~':
+						normalized ~= conv; // Decode unreserved
+						break;
+					default:
+						normalized ~= input[i .. i + 3].toUpper(); // Uppercase HEX
+						break;
+				}
+
+				i += 2;
+			}
+			else
+				normalized ~= c;
+		}
+
+		return normalized.idup;
```

Comment:
```suggestion
		return normalized.assumeUnique;
```

Or use an `Appender` :)

Reply:
`assumeUnique` is not `@safe`, thus using appender